### DISCUSSION TITLE
NAT Loopback

### DISCUSSION THREAD

**Discussion by ITFS-tech** [rustdesk/rustdesk-server-pro#440] [General]
Hello everyone.

I have installed RustDesk Pro Server on a Windows 10 Pro VM on Proxmox. Everything is running fine, but I'm struggling with the NAT loopback and would just love to be able to reach my clients on the local network from the outside.

What I have tried:
-PiHole (has always been running with me for a long time), after entering a DNS masking nothing worked at all, even connections to remote servers did not work anymore
-Hosts file adjusted (IP of my server redirected to my domain rustdesk.something.com, entry is “172.178.2.50 rustdesk.someting.com”)
-DNS rebind in my router, entered exactly as in the hosts file

All ports from outside are open as specified. How can I find out why it is not working? What else can I try? Does a relay server help here?

Thanks for your help in advance!

**Comment by risaw1981** [comment]
Hey

I have a similar setup, using ESXI, my firewall is a PfSense, took me  while to figure it out but you need to look at DNS Resolver and enter some "Host Overrides"

Obviously if you are using a different DNS server or firewall you'll have to look for the alternative jargon but on PfSense its "DNS Resolver" and "Host Overrides"

Basically enter your Rust Desk URL and point it to the local IP

**Comment by risaw1981** [comment]
Before doing this I was just editing the C:\Windows\System32\drivers\etc\hosts file if it was a Windows machine and creating a DNS entry in there but its not ideal obviously.

**Reply by ITFS-tech** [reply]
Hello risaw1981.

Thank you for your support.

as i have already written in my post, the hosts file has already been edited. but it did not work.

my dns resolution is done by pihole and was implemented exactly as described in the rustdesk documentation. that didn't work either.

**Reply by risaw1981** [reply]
In your original post you mention it working ok on a WIN10 VM, does that mean the anything on the Proxmox LAN works OK, its outside of the Proxmox LAN, it doesn't work?

**Reply by ITFS-tech** [reply]
No. Unfortunately, you misunderstood. Everything works perfectly, I just can't access my other internal devices from outside via Rustdesk.

**Reply by risaw1981** [reply]
Ok got ya, I setup my RustDesk self hosted as per instructions and found I had to edit the API server to https instead of the autogenerated http. Do you get any error code or info?

**Comment by ITFS-tech** [comment]
I have now tried that too. It looks good at first, but access from outside only works as long as I'm not logged in.

So if the API in the Rustdesk client is set with https (although I don't use an SSL certificate on the server), then access from outside works. But then the login to the server no longer works and I can neither access the address book nor the favorites. So I can either access my internal devices from outside or control external devices from here, but neither works.

**Comment by ITFS-tech** [comment]
Actually what about using a public relay server from rustdesk community? This should solve the problem also?!?

**Reply by risaw1981** [reply]
Worth a  go I suppose, if all else fails, speak with RustDesk directly via email
